### PR TITLE
fix repo url being decoded while downloading repo index

### LINKED ISSUE #6601: Helm3 should not decode URLs
Related helm v2 issue - #6048 

Output of `helm version`:
```
version.BuildInfo{Version:"v3.0+unreleased", GitCommit:"cc450563835088945aa79dbfa0bce0bc031fb390", GitTreeState:"clean", GoVersion:"go1.13.1"}
```

```
$ helm3 repo add blah http://localhost:8080/blah%2Fpath/blah
Error: looks like "http://localhost:8080/blah%2Fpath/blah" is not a valid chart repository or cannot be reached: Get http://localhost:8080/blah/path/blah/index.yaml: dial tcp [::1]:8080: connect: connection refused
```

### PR DESCRIPTION
**Pull request by karuppiah7890**
**What this PR does / why we need it**:
Port of #6060 (v2 fix)

**If applicable**:
- [ ] this PR contains documentation
- [X] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

**Review by hickeyma — APPROVED** [review]
LGTM, thanks @karuppiah7890

**Comment by hickeyma** [comment]
Needs another maintainer lgtm to merge.

**Review by bacongobbler — APPROVED** [review]

**Pull request merged**
